### ISSUE TITLE
Can we remove support for pyfits now (i.e. just support astropy.io.fits)?

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#460]
We have a little bit of code that supports different versions of pyfits, which is untested. I think it's a good time to drop pyfits support (we've labelled it as deprecated for a while now) and just support astropy.io.fits.

**Comment by DougBurke** [comment]
In `sherpa.astro.io.pyfits_backend.py` we have

```
try:
    from astropy.io import fits
except ImportError:
import pyfits as fits

...

# Should we drop support for earlier versions? The code is only tested
# using recent (3.3 or later) versions, so there are no guarantees
# for earlier versions. Should the import order below be reversed,
# so that later versions are tried first?
#
try:
    # pyfits-1.3 support
    _VLF = fits.NP_pyfits._VLF
except AttributeError:
    # pyfits-2.3.1 support
    try:
        _VLF = fits.core._VLF
    except AttributeError:
        # pyfits-3.0 support
        _VLF = fits.column._VLF

# fits.CardList is deprecated
if hasattr(fits, 'Header'):
    _new_header = fits.Header
else:
    _new_header = fits.CardList


# As fits.new_table is deprecated, use the from_columns method
# if available. Note that this should only be done for binary
# tables, as fits.TableHDU.from_columns should be used for
# ASCII tables.
#
if hasattr(fits.BinTableHDU, "from_columns"):
    _new_table = fits.BinTableHDU.from_columns
else:
_new_table = fits.new_table
```

The `Header`/`from_columns` change is - I believe, "relatively" new (i.e. it changed in `astropy`) - so this would be kept (although should add in the version of `astropy` when the change was made so we know when to drop this code). Hopefully we can remove the various attempts to find the `_VLF` symbol: we just need to check that the versions of astropy we want to support support this.

**Comment by DougBurke** [comment]
I installed astropy version 1.0.6 from anaconda (note that I believe AstroPy v 1.0 is now consider "unsupported" but I couldn't find an official document to point to here), and note that it has `astropy.io.fits.column._VLF` and not the `core` or `NP_pyfits` variants, as shown below. So I'll put together a simple PR to clean up the imports here.

```
% python
Python 2.7.15 |Anaconda, Inc.| (default, May  1 2018, 23:32:55) 
[GCC 7.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from astropy.io import fits
>>> import astropy
>>> astropy.__version__
u'1.0.6'
>>> fits.column._VLF
<class 'astropy.io.fits.column._VLF'>
>>> fits.core._VLF
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'core'
>>> fits.NP_pyfits._VLF
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'NP_pyfits'
```

**Comment by DougBurke** [comment]
See #462 for an initial attempt at this.

**Comment by olaurino** [comment]
Closed by #462
